feedback/migrations/0020_remove_feedbacktag_feedbacks_conversation_and_more.py

```diff
@@ -0,0 +1,78 @@
+# Generated by Django 4.2.1 on 2023-07-06 09:48
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+def generate_conversations_for_old_feedback(apps, schema_editor):
+    Conversation = apps.get_model("feedback", "Conversation")
+    Feedback = apps.get_model("feedback", "Feedback")
+    for fb in Feedback.objects.all():
+        conv, _created = Conversation.objects.get_or_create(
+            exercise=fb.exercise,
+            student=fb.student,
+        )
+        fb.conversation = conv
+        fb.save()
+
+
+def refer_tag_to_conversation(apps, schema_editor):
+    FeedbackTag = apps.get_model("feedback", "FeedbackTag")
+    for tag in FeedbackTag.objects.all():
+        tag.conversations.add(*[fb.conversation for fb in tag.feedbacks.all()])
```

Comment:
@etanttila The migrations are causing trouble in the production server because they consume too much memory. Perhaps this line increases the memory consumption too much. I suspect that the list comprehension `[... for .. in ...]` forces Python to load all records from `tag.feedbacks.all()` into memory at once. If this function were implemented differently, then hopefully, the memory consumption could be reduced drastically.

Reply:
Good point, or perhaps it might be better to use [only()](https://docs.djangoproject.com/en/4.2/ref/models/querysets/#django.db.models.query.QuerySet.only), since probably only the id would be needed in this case. I also now realized that one could also use `set` rather than `add`, in which case one can provide a list as the argument, which would remove the need for the spread operation. I'll test locally if using `only` and `set` works.